lib/gi/client.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+require 'common/client/base'
+require 'gi/configuration'
+
+module Gi
+  # Core class responsible for api interface operations
+  class Client < Common::Client::Base
+    configuration Gi::Configuration
+
+    def get_autocomplete_suggestions(term)
+      json_hash = perform(:get, 'institutions/autocomplete', {term: term}, nil).body
```

Comment:
@omgitsbillryan the Faraday middle-ware defined in lib/gi/configuration for "raise_error" should be able to handle timeouts. An appropriate error would be rendered at that level and bubble up to application controller.